app/services/SubmissionService.scala

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import connectors.IvdSubmissionConnector
+import models.DocumentTypes.{DefermentAuthorisation, DocumentType}
+import models.OptionalDocument._
+import models.SelectedDutyTypes.{Duty, Vat}
+import models.requests.DataRequest
+import models._
+import play.api.Logger
+import play.api.libs.json.{JsError, JsObject, JsSuccess, JsValue, Json}
+import uk.gov.hmrc.http.HeaderCarrier
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class SubmissionService @Inject()(ivdSubmissionConnector: IvdSubmissionConnector) {
+
+  private val logger = Logger("application." + getClass.getCanonicalName)
+
+  def createCase()(implicit request: DataRequest[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Either[ErrorModel, SubmissionResponse]] = {
+
+    buildSubmission(request.userAnswers) match {
+      case Right(submission) => {
+        ivdSubmissionConnector.createCase(submission).map {
+          case Right(confirmationResponse) => Right(confirmationResponse)
+          case Left(errorResponse) => {
+            logger.error("Error in response from EIS/Pega: " + errorResponse.message)
+            Left(errorResponse)
+          }
+        }
+      }
+      case Left(err) => Future.successful(Left(err))
+    }
+
+  }
+
+  private[services] def buildSubmission(answers: UserAnswers): Either[ErrorModel, JsValue] = {
+    Json.fromJson[SubmissionData](answers.data) match {
+      case JsSuccess(data, _) =>
+        Right(
+          buildEntryDetails(data) ++
```

Comment:
Can the `buildX` functions be refactored to return `Either[BuildError, JsObject]` so errors building the sections can be logged? At present some `buildX` functions return an empty JSON object when something goes wrong which is not what we want.

Reply:
done